src/main/java/ee/tuleva/onboarding/comparisons/fundvalue/retrieval/GlobalStockValueRetriever.java

```diff
@@ -0,0 +1,64 @@
+package ee.tuleva.onboarding.comparisons.fundvalue.retrieval;
+
+import ee.tuleva.onboarding.comparisons.fundvalue.FundValue;
+import ee.tuleva.onboarding.comparisons.fundvalue.persistence.FundValueRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class GlobalStockValueRetriever implements ComparisonIndexRetriever {
+    public static final String KEY = "NEW_GLOBAL_STOCK_INDEX";
```

Comment:
Maybe change the name to something like `UNION_STOCK_INDEX` and the class name to 
`UnionStockIndexRetriever` (together with the test).